strawberryfields/io.py

```diff
@@ -146,6 +176,71 @@ def to_program(bb):
     return prog
 
 
+# pylint:disable=too-many-branches
+def _to_tdm_program(bb):
+    # pylint: disable=import-outside-toplevel
+    from strawberryfields.tdm.tdmprogram import TDMProgram
+
+    prog = TDMProgram(max(bb.modes) + 1, name=bb.name)
+
+    # retrieve all the free parameters in the Blackbird program (e.g. "p0", "p1"
+    # etc.) and add their corresponding values to args
+    args = []
+    for k in bb._var.keys():
+        if k[0] == "p" and k[1:].isdigit():
```

Comment:
Could create something like `is_bb_free_param = lambda param: isinstance(param, str) and param[0] == "p" and param[1:].isdigit()` to help with readability and usage here and at later occurences.